src/components/tutorial-card/helpers/with-auth-elements.tsx

```diff
@@ -36,6 +42,14 @@ export function TutorialCardWithAuthElements(props: TutorialCardPropsWithId) {
 		parseInt(progressStatusToPercent(TutorialProgressStatus.in_progress))
 	const shouldRenderProgress = isProgressLoaded && meetsProgressThreshold
 
+	/**
+	 * In certain contexts, we want the bookmark button to render
+	 * with a confirmation dialog box.
+	 */
+	const BookmarkButton = renderBookmarkConfirmationDialog
```

Comment:
If `renderBookmarkConfirmationDialog` is truthy, we use `BookmarkButtonWithRemoveDialog`.

Note: could probably make sense to move `BookmarkButtonWithRemoveDialog` into `components/bookmark-button`. Left it in place for now for cleaner diff in initial review, happy to move if this approach feels acceptable overall.

Reply:
Note: have refactored to accept a `BookmarkButtonComponent` prop instead 👍